src/boot/efi/boot.c

```diff
@@ -812,12 +828,30 @@ static BOOLEAN menu_run(
                 case KEYPRESS(0, 0, 'p'):
                 case KEYPRESS(0, 0, 'P'):
                         print_status(config, loaded_image_path);
-                        refresh = TRUE;
+                        clear = TRUE;
                         break;
 
                 case KEYPRESS(EFI_CONTROL_PRESSED, 0, 'l'):
                 case KEYPRESS(EFI_CONTROL_PRESSED, 0, CHAR_CTRL('l')):
-                        refresh = TRUE;
+                        clear = TRUE;
+                        break;
+
+                case KEYPRESS(0, 0, 'r'):
+                case KEYPRESS(0, 0, 'R'):
+                        if (config->console_mode_efivar < ST->ConOut->Mode->MaxMode &&
+                            ST->ConOut->Mode->Mode == ST->ConOut->Mode->MaxMode - 1LL) {
+                                config->console_mode_efivar = CONSOLE_MODE_KEEP;
```

Comment:
as mentioned earlier, why not save  the `ST->ConOut->Mode->Mode` value we found when we initially started in some variable, and then issue console_set_mode() here, back to that?